iTechArt.Common/Helpers/Logger/NLogger.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.ComponentModel;
+using NLog;
+namespace iTechArt.Common.Helpers.Logger
+{
+    public class NLogger : ILoggerBase
```

Comment:
Right now you have added dependency to NLogger implementation and their interfaces. What if I want to move to Log4Net?

It's much better to create in Common our own interface:

```
public interface ILogger
{
   void Log(...);

   void LogWarning(...);

   void LogError(...);
}
```
Then create separate project that will contain implementation that is based on NLogger.

There is Ambient Context pattern exist that is commonly used for such things like logging. Logging is used across all projects and we may want to avoid injecting logger in each class. So, we may implement the following Context class:

```
public static class LoggerContext
    {
        private static ILogger _current;
        private static readonly ILogger DefaultLogger = new ...();


        public static ILogger Current
        {
            get
            {
                return _current ?? (_current = DefaultLogger);
            }
            set
            {
                _current = value;
            }
        }
    }
```

The idea is pretty simple: on application initialization(or even in rutime) we could setup Current and replace current(default) logger. And then logging in your Controller will look like: 

LoggerContext.Current.Log(...);